apps/src/gamelab/GameLabView.jsx

```diff
@@ -0,0 +1,81 @@
+/** @file Top-level view for GameLab */
+'use strict';
+
+var connect = require('react-redux').connect;
+var ConnectedStudioAppWrapper = require('../templates/ConnectedStudioAppWrapper.jsx');
+var GameLabInterfaceMode = require('./constants').GameLabInterfaceMode;
+var GameLabVisualizationHeader = require('./GameLabVisualizationHeader.jsx');
+var ProtectedStatefulDiv = require('../templates/ProtectedStatefulDiv.jsx');
+
+/**
+ * Top-level React wrapper for GameLab
+ */
+var GameLabView = React.createClass({
+  propTypes: {
+    interfaceMode: React.PropTypes.bool.isRequired,
+    isEmbedView: React.PropTypes.bool.isRequired,
+    isShareView: React.PropTypes.bool.isRequired,
+    renderCodeWorkspace: React.PropTypes.func.isRequired,
+    renderVisualizationColumn: React.PropTypes.func.isRequired,
+    onMount: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount: function () {
+    this.props.onMount();
+  },
+
+  render: function () {
+    return (
+      <ConnectedStudioAppWrapper>
+        {this.renderCodeMode()}
+        {this.renderAnimationMode()}
+      </ConnectedStudioAppWrapper>
+    );
+  },
+
+  renderCodeMode: function () {
+    // Code mode contains protected (non-React) content.  We have to always
+    // render it, so when we're not in code mode use CSS to hide it.
+    var codeModeStyle = {};
+    if (this.props.interfaceMode !== GameLabInterfaceMode.CODE) {
+      codeModeStyle.display = 'none';
+    }
```

Comment:
I think how you have it in this comment is how I would have written, but as I think about this, I dont feel strongly about it at all. I'm fine with either approach.